app/controllers/markets_controller.rb

```diff
@@ -1,10 +1,53 @@
 class MarketsController < ApplicationController
 
   def sign_in
-
+    @title = "Farmers Market Sign In"
   end
 
   def index
+    @markets = Market.all
+    @title = "Markets"
+  end
+
+  def show
+    id = params[:id]
+    @market = Market.find(id)
+    @title = "Market Info"
+  end
+
+  def new
+    @market = Market.new()
+    @action = "create"
+    @title = "Add a Market"
+  end
+
+  def create
+    Market.create(market_params[:market])
+    redirect_to "/markets"
+  end
+
+  def edit
+    id = params[:id]
+    @market = Market.find(id)
```

Comment:
I think it would be great to see the `id` and the `@market` variable assignment moved into a `before_action` for these specific methods.